Resources/Computing/ComputingElement.py

```diff
@@ -116,7 +116,12 @@ def initializeParameters( self ):
     """
 
     # Collect global defaults first
-    for section in [ '/Resources/Computing/CEDefaults', '/Resources/Computing/%s' % self.ceName ]:
+    siteName = gConfig.getValue( '/LocalSite/Site' )
+    siteType = siteName.split( '.' )[0]
+    if not self.ceName:
+      self.ceName = gConfig.getValue( '/LocalSite/GridCE' )
+    for section in [ '/Resources/Computing/CEDefaults', '/Resources/Computing/%s' % self.ceName, '/Resources/Sites/%s/%s/CEs/%s' %
+                                                                                                 ( siteType, siteName, self.ceName )  ]:
```

Comment:
/Resources/Computing is used to communicate extra resource description parameters through the local configuration, e.g. by setting the pilot arguments. However, the ComputingElement is supposed to get its parameters while creation by the Factory and not discover it from the environment. otherwise the Factory pattern does not make sense